help/develop/domains/chem/cheminformatics.md

```diff
@@ -31,59 +31,36 @@ descriptors(table, column, descriptors);
+If i-th element of the input `column` contains a substructure given by a `pattern` the i-th bit set to 1, otherwise the bit is set to 0.
 
-`column` is a column of type String containing molecules, each in any notation
-RDKit-JS [supports](https://github.com/rdkit/rdkit/blob/master/Code/MinimalLib/minilib.h): smiles, cxsmiles, molblock,
-v3Kmolblock, and inchi. Same applies to a `pattern`: this can be a string representation of a molecule in any of the
-previous notations.
+`column` is a column of type String. It contains molecules in any notation [supported by RDKit JS](https://github.com/rdkit/rdkit/blob/master/Code/MinimalLib/minilib.h): smiles, cxsmiles, molblock, v3Kmolblock, and inchi. Same applies to a `pattern`: this is a string representation of a molecule in any of the previous notations.
 
 The `settings` object allows passing the following parameters:
 
-* `substructLibrary`: a boolean indicating whether to use the "naive" graph-based search method,
-  using [`get_substruct_match`](https://www.rdkit.org/docs/source/rdkit.Chem.rdchem.html) method of
-  RDKit (`substructLibrary: false`
-  , [code sample 1](https://public.datagrok.ai/js/samples/domains/chem/substructure-search-simple)
-  , [code sample 2](https://public.datagrok.ai/js/samples/domains/chem/substructure-search-library))
-  , or a [`substructLibrary`](https://rdkit.blogspot.com/2018/02/introducing-substructlibrary.html)
-  from RDKit (`substructLibrary: true`
-  , [code sample](https://public.datagrok.ai/js/samples/domains/chem/substructure-search-library)). The default value
-  is `true`.
-
-For the chosen `substructLibrary` option, the first call to the function with a `column` previously unseen by
-the `searchSubstructure` function shall build a library. The library is used for further searches in the same given
-column while the `pattern` is varying.
-
-The function's user doesn't need to manage the built library, as it is done by the function automatically. As long as
-the function detects it has received a column different to the one previously seen, it shall rebuild the library for
-this newly seen column before continuing with searches.
-
-It is possible to call `searchSubstructure` solely for initializing the library, without performing an actual search. In
-order to do this, `pattern` should be passed with a value of either `null` or an empty string `'''`.
-
-Sometimes it happens that the molecule strings aren't supported by RDKit-JS. These inputs are skipped when indexing, but
-an error log entry is added for each such entry to the browser's console together with the actual molecule string. For
-such unsupported molecule, the corresponding bit in the bitset remains as 0. Thus, the bitset is *always* of the
+* `molBlockFailover`: smarts which used as a substructure if `pattern` is invalid
+
+To make search maximum efficient substructure search function maintains a cache of
```

Comment:
Do you mean `To make search as effcient as possible/more efficient`? 
If the answer is yes, we should modify the sentence to make it grammatically correct.
`To optimize the substructure search, we use a cache of pre-computed etc... `